quickstart/kubernetes-guestbook.md

```diff
@@ -0,0 +1,373 @@
+     +   ├─ kubernetes:core:Service     redis-slave                      created     1 info message
+     +   ├─ kubernetes:core:Service     frontend                         created     1 info message
+     +   ├─ kubernetes:core:Service     redis-master                     created     1 info message
+     +   ├─ kubernetes:apps:Deployment  redis-master                     created
+     +   ├─ kubernetes:apps:Deployment  redis-slave                      created
+     +   └─ kubernetes:apps:Deployment  frontend                         created
+
+    Diagnostics:
+      kubernetes:core:Service: redis-slave
+        info: ✅ Service 'redis-slave' successfully created endpoint objects
+
+      kubernetes:core:Service: frontend
+        info: ✅ Service 'frontend' successfully created endpoint objects
+
+      kubernetes:core:Service: redis-master
+        info: ✅ Service 'redis-master' successfully created endpoint objects
+
+    ---outputs:---
+    frontendIP: "10.102.193.86"
+
+    info: 7 changes performed:
+        + 7 resources created
+    Update duration: 16.226520447s
+
+    Permalink: https://app.pulumi.com/joeduffy/k8s-guestbook-dev/updates/1
+    ```
+
+4.  The application is now running in our Kubernetes cluster.
+
+    Use `kubectl` to see the deployed services:
+
+    ```shell
+    $ kubectl get services
+    ```
+
+    We should see entries for each of the four Services we've created in our program:
+
+    ```
+    NAME           TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
+    frontend       ClusterIP   10.102.193.86   <none>        80/TCP     2m
+    kubernetes     ClusterIP   10.96.0.1       <none>        443/TCP    1d
+    redis-master   ClusterIP   10.98.205.37    <none>        6379/TCP   2m
+    redis-slave    ClusterIP   10.96.9.70      <none>        6379/TCP   2m
+    ```
+
+    The `pulumi stack output` command prints exported program variables:
+
+    ```shell
+    $ pulumi stack output frontendIP
+    ```
+
+    The value of the `frontendIP` variable matches exactly the `frontend`'s `CLUSTER-IP` reported by `kubectl`:
+
+    ```
+    10.102.193.86
+    ```
+
+    Because Minikube doesn't support the `LoadBalancer` type, our example above uses `ClusterIP`. In order to
+    browse to it, we will first need to forward a port on `localhost` to it. To do so, run:
+
+    ```shell
+    $ kubectl port-forward svc/frontend 8765:80
+    ```
+
+    At this point, we can view our running Guestbook application:
+
+    ```shell
+    $ curl localhost:8765
+    ```
+
+    The HTML from the Guestbook will be fetched and printed:
+
+    ```
+    <html ng-app="redis">
+      <head>
+      <title>Guestbook</title>
+      ...
+    </html>
+    ```
+
+    If you are instead running this program in a real cluster, you can change `ClusterIP` to `LoadBalancer` above,
+    run `pulumi up` to update it, and then you can simply access your Guestbook application with:
+
+    ```shell
+    $ curl $(pulumi stack output frontendIP)
+    ```
```

Comment:
I actually like the idea of putting all this logic in the code as a config var (as the examples repo version of this does), so that it is in code instead of in prose.  I feel like it's an opportunity to comment on how it is easy to make the deployment adaptive via this configuration?

Reply:
I'm fine doing that, but I'm going to flip the polarity (and think we should on the example also). Basically, assume there is no load balancer unless explicitly stated; this is simpler, and for the newbie users that are relying on Minikube, it's one less thing to grok when you're just trying to get up and running. Presumably folks with a real cluster will actually know they need a load balancer, whereas these newer users won't know they don't need one.